src/screens/WalletType.js

```diff
@@ -0,0 +1,234 @@
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the get version call
+   */
+  handleVersion = (event, arg) => {
+    if (arg.success) {
+      // We wait 2 seconds to update the message on the screen
+      // so the user don't see the screen updating fast
+      setTimeout(() => {
+        this.setState({ waitAction: true, errorMessage: '' }, () => {
+          ledger.getPublicKeyData();
+        })
+      }, 2000);
+    } else {
+      if (!this.state.hardware) {
+        // User clicked 'Back' button
+        this.attemptNumber = 0;
+        return;
+      }
+
+      if (this.attemptNumber < this.attemptLimit) {
+        setTimeout(() => {
+          this.attemptNumber += 1;
+          ledger.getVersion();
+        }, this.attemptInterval);
+      } else {
+        // Error
+        this.setState({ errorMessage: arg.error.message });
+      }
+    }
+  }
+
+  /**
+   * Handle the response of a get public key data call to Ledger.
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the get public key data call
+   */
+  handlePublicKeyData = (event, arg) => {
+    if (!this.state.hardware) {
+      // User clicked 'Back' button
+      return;
+    }
+
+    if (arg.success) {
+      const data = arg.data;
+      const uncompressedPubkey = data.slice(0, 65);
+      const compressedPubkey = hathorLib.wallet.toPubkeyCompressed(uncompressedPubkey);
+      const chainCode = Buffer.from(data.slice(65, 97));
+      const fingerprint = Buffer.from(data.slice(97, 101));
+      const xpub = hathorLib.wallet.xpubFromData(compressedPubkey, chainCode, fingerprint);
+      // First we clean what can still be there of a last wallet
+      wallet.cleanWallet();
+      wallet.startHardwareWallet(xpub);
+      hathorLib.wallet.markBackupAsDone();
+      this.props.history.push('/wallet/');
+    } else {
+      // Error
+      this.setState({ errorMessage: arg.error.message });
+    }
+  }
+
+  /**
+   * Go to software wallet warning screen
+   */
+  goToSoftwareWallet = () => {
+    hathorLib.wallet.setWalletType('software');
+    this.props.history.push('/software_warning/');
+  }
+
+  /**
+   * Go to the hardware wallet screen
+   */
+  goToHardwareWallet = () => {
+    hathorLib.wallet.setWalletType('hardware');
+    this.setState({ hardware: true }, () => {
+      this.attemptNumber = 1;
+      ledger.getVersion();
+    });
+  }
+
+  /*
+   * Try getting user approval on Ledger again
+   */
+  tryAgain = () => {
+    // clear error
+    this.setState({ errorMessage: null, waitAction: false }, () => {
+      this.attemptNumber = 1;
+      ledger.getVersion();
+    });
+  }
+
+  render() {
+    const renderInitial  = () => {
+      return (
+        <div>
+          <p className="mt-4 mb-4">{t`Do you want to connect to a hardware device or to start a software wallet?`}</p>
```

Comment:
On #52 @msbrogli suggested improving the message with a better explanation about "Hardware vs Software" wallets. @msbrogli what do you have in mind?

Reply:
Maybe something like: 

"""
Hathor Wallet supports two types of wallet: software and hardware. The primary difference between them is where your seed is stored. While a software wallet stores your seed in your computer, the hardware wallet stores it in a separate hardware.

Even though it is simpler and faster to use a software wallet, your seed may be stolen and transmitted through the internet by a virus.

The most important benefit of using a hardware wallet is to keep your seed safe. No virus can stole it from the hardware.

For further information, check out [the Software vs Hardware Wallet Comparison page](...).

Which type of wallet would you like to use?
"""

I think we have already put a brief text similar to this one. Can you double check, please?

What do you think?

Reply:
We display this text *after* someone selects the Software Wallet:

"Using a software wallet is not the safest way to secure your tokens. As your seed phrase is stored in your computer and it is connected to the internet, your seed phrase may be stolen by a virus.

If you want a safer method, you can use a hardware wallet. We currently have support for Ledger."

Reply:
Ok. Maybe we can put just the first paragraph with a simple explanation (and no comparison) of each type of wallet.

"""
Hathor Wallet supports two types of wallet: software and hardware. The primary difference between them is where your seed is stored. While a software wallet stores your seed in your computer, the hardware wallet stores it in a separate hardware.
"""

What do you think?

Reply:
I think people who are not familiar with hardware wallets won't understand it and become even more confused.

> the hardware wallet stores it in a separate hardware

Is it a separate hardware in the same computer? I don't think most people will understand that's an external, dedicated hardware. Also, they probably won't understand what a seed means.

"Hathor Wallet supports two types of wallet: software and hardware.

**Hardware wallets** are dedicated external devices that store your private information. We currently support the Ledger hardware wallet. **Software wallets**, on the other hand, store the information on your computer."

It'd be good to have a link further explaining the difference. We could have a page on our site or link to an external one, like [this](https://medium.com/coincontrolblog/hardware-wallets-vs-software-wallets-e817b54006b4).

Reply:
> """
> Hathor Wallet supports two types of wallet: software and hardware.
> 
> **Hardware wallets** are dedicated external devices that store your private information. We currently support the Ledger hardware wallet. **Software wallets**, on the other hand, store the information on your computer.
> """

Great explanation! I think we should use it. I also agree that we should add a page explaining it in our website.

Reply:
Done. Also created an issue on the website repo to create this page.